lib/collection/src/shards/replica_set.rs

```diff
@@ -890,9 +917,19 @@ impl ShardReplicaSet {
                 Ok(Res::resolve(responses, condition))
             }
         } else {
-            Err(CollectionError::service_error(
-                "Failed to complete read operation: too many replicas returned an error".into(),
-            ))
+            let success_count = responses.len();
+            let error_count = errors.len();
+
+            Err(CollectionError::service_error(format!(
+                "{} of {} shards failed with: {}",
+                error_count,
```

Comment:
Nitpick: we can add `error_count` in the format string.

Nice use of `join` below!

Reply:
copilot suggested